### PR TITLE
Feature - Add poppler_path argument for specify where to look for poppler binaries

### PR DESCRIPTION
**Pull request by bratao**
Hello,

Thank you for this very handy library.

I need to be able to specify where to look for Poppler binaries. I added a helper function and a new argument to be able to specify where to look.

### CODE REVIEW

File: pdf2image/pdf2image.py [hunk 1]
```diff
@@ -26,12 +27,16 @@
 
 TRANSPARENT_FILE_TYPES = ['png', 'tiff']
 
-def convert_from_path(pdf_path, dpi=200, output_folder=None, first_page=None, last_page=None, fmt='ppm', thread_count=1, userpw=None, use_cropbox=False, strict=False, transparent=False, output_file=str(uuid.uuid4())):
+
+def convert_from_path(pdf_path, dpi=200, poppler_path=None, output_folder=None, first_page=None, last_page=None,
```
Comment (Belval) [inline pdf2image/pdf2image.py]: Please add the new parameter last, while I understand that it's not as simply semantically, it prevents backward incompatible changes if users were not using the named parameters.

File: pdf2image/pdf2image.py [hunk 2]
```diff
@@ -105,17 +110,22 @@ def convert_from_path(pdf_path, dpi=200, output_folder=None, first_page=None, la
         else:
             images += parse_buffer_func(data)
 
+    for uid, proc in processes:
```
Comment (Belval) [inline pdf2image/pdf2image.py]: What does this achieve?

File: pdf2image/pdf2image.py [hunk 3]
```diff
@@ -105,17 +110,22 @@ def convert_from_path(pdf_path, dpi=200, output_folder=None, first_page=None, la
...
+        proc.kill()
+
     if auto_temp_dir:
         shutil.rmtree(output_folder)
 
     return images
 
-def convert_from_bytes(pdf_file, dpi=200, output_folder=None, first_page=None, last_page=None, fmt='ppm', thread_count=1, userpw=None, use_cropbox=False, strict=False, transparent=False, output_file=str(uuid.uuid4())):
+
+def convert_from_bytes(pdf_file, dpi=200, poppler_path=None, output_folder=None, first_page=None, last_page=None, fmt='ppm', thread_count=1, userpw=None, use_cropbox=False, strict=False, transparent=False, output_file=str(uuid.uuid4())):
```
Comment (Belval) [inline pdf2image/pdf2image.py]: Same as above, new parameter last.

File: pdf2image/pdf2image.py [hunk 4]
```diff
@@ -132,11 +142,15 @@ def convert_from_bytes(pdf_file, dpi=200, output_folder=None, first_page=None, l
...
-            return convert_from_path(f.name, dpi=dpi, output_folder=output_folder, first_page=first_page, last_page=last_page, fmt=fmt, thread_count=thread_count, userpw=userpw, use_cropbox=use_cropbox, strict=strict, transparent=transparent, output_file=output_file)
+            return convert_from_path(f.name, dpi=dpi, poppler_path=poppler_path, output_folder=output_folder,
+                                     first_page=first_page, last_page=last_page, fmt=fmt, thread_count=thread_count,
+                                     userpw=userpw, use_cropbox=use_cropbox, strict=strict, transparent=transparent,
+                                     output_file=output_file)
     finally:
         os.close(fh)
         os.remove(temp_filename)
 
+
```
Comment (Belval) [inline pdf2image/pdf2image.py]: Adding a line here respects the PEP8, thank you for that.

File: pdf2image/pdf2image.py [hunk 5]
```diff
@@ -174,12 +189,27 @@ def _parse_format(fmt):
     # Unable to parse the format so we'll use the default
     return 'ppm', parse_buffer_to_ppm, False
 
-def _page_count(pdf_path, userpw=None):
+
+def _get_command_path(command, poppler_path=None):
+    command = command
```
Comment (Belval) [inline pdf2image/pdf2image.py]: I believe this declaration is redundant.

File: pdf2image/pdf2image.py (hunk 2 above)
Comment (bratao) [reply pdf2image/pdf2image.py]: In Windows, sometimes the Poppler process are still active holding the handle for the input file. If try to remove the file immediately after pdf2image call, I fails.

**Comment by bratao** [comment]
Thank you Belval. 
Could you please check now?

**Comment by Belval** [comment]
Your changes are good to go, I'll test on my side and merge everything tonight.

File: pdf2image/pdf2image.py [hunk 6]
```diff
@@ -174,12 +192,25 @@ def _parse_format(fmt):
...
+
+def _page_count(pdf_path, userpw=None, poppler_path=None):
     try:
+        command = [_get_command_path("pdfinfo", poppler_path), pdf_path]
+
         if userpw is not None:
-            proc = Popen(["pdfinfo", pdf_path, '-upw', userpw], stdout=PIPE, stderr=PIPE)
-        else:
-            proc = Popen(["pdfinfo", pdf_path], stdout=PIPE, stderr=PIPE)
+            command = command + [userpw]
```
Comment (Belval) [inline pdf2image/pdf2image.py]: This does not work, see https://travis-ci.org/Belval/pdf2image/jobs/509002906

You need to add the `-upw` parameter.

**Comment by bratao** [comment]
@Belval , can you check it now? All tests passed!
Thank you for your patience

**Comment by Belval** [comment]
Nice work! I'll release version 1.5 in a few hours.

**Pull request merged**